kv_server_grpc.cc

```diff
@@ -1,51 +1,50 @@
+#include "kv_server_grpc.h"
+
 #include <iostream>
-#include <memory>
-#include <string>
-
-#ifdef BAZEL_BUILD
-// TODO: I'm not sure what to put here in place of the proto stuff that hello world uses
-// #include "examples/protos/helloworld.grpc.pb.h"
-// #else
-// #include "helloworld.grpc.pb.h"
-#else
-
-#endif
-
-// Key Value Server part of grpc that makes calls to the true kv_backend
-class KeyValueServer final : public chirp::KeyValueStore::Service {
-  public:
-    Status Put(ServerContext* context, const chirp::PutRequest* request, chirp::PutReply* reply) override {
-      // TODO: find out how to generate random keys
-      int key = 0;
-      kvbe.Put(key, request);
-      return Status::OK;
-    }
-
-    Status Get(ServerContext* context, const chirp::GetRequest* request, chirp::GetReply* reply) override {
-      *reply = kvbe.Get(request);
-      return Status::OK;
-    }
-
-    Status DeleteKey(ServerContext* context, const chirp::DelteKeyRequest* request, chirp::DeleteKeyReply* reply) override {
-      kvbe.DeleteKey(request);
-      return Status::OK;
-    }
-  private:
-    KeyValueBackEnd kvbe;
-};
+
+grpc::Status KeyValueServer::put(grpc::ServerContext* context, const chirp::PutRequest* request, chirp::PutReply* reply) {
+  //unwrap the request's fields so that we may pass them to the ServiceLayer's data structure
+  std::string key = request->key();
+  std::string value = request->value();
+  // call KeyValueBackend's Put function
+  key_value_back_end_.Put(key, value);
+  return grpc::Status::OK;
+}
+
+grpc::Status KeyValueServer::get(grpc::ServerContext* context, grpc::ServerReaderWriter<chirp::GetReply, chirp::GetRequest>* stream) {
+  //unwrap the request's fields so that we may pass them to the ServiceLayer's data structure
+  chirp::GetRequest request;
+  stream->Read(&request);
+  // call KeyValueBackend's Get function
+  const std::vector<std::string>& kValues = key_value_back_end_.Get(request.key());
+
+  chirp::GetReply reply;
+  for (const std::string& kVal : kValues) {
+    reply.set_value(kVal);
+    const chirp::GetReply& kSendingReply = reply;
+    stream->Write(kSendingReply);
+  }
+  return grpc::Status::OK;
+}
+
+grpc::Status KeyValueServer::deletekey(grpc::ServerContext* context, const chirp::DeleteRequest* request, chirp::DeleteReply* reply) {
+  // call KeyValueBackend's DeleteKey function
+  key_value_back_end_.DeleteKey(request->key());
+  return grpc::Status::OK;
+}
 
 void RunServer() {
```

Comment:
I would separate this into a different file, this should be just for class methods